In [ ]:
#

# Q5. A survey was conducted to examine the relationship between age and preference for a particular
# brand of soft drink. The survey results are shown below:
# Age(Years)
# 25 Coke
# 42 Pepsi
# 37
# 19
# 31
# 28

# Mountain dew
# Coke
# Pepsi
# Coke

# Q6. A company is interested in examining the relationship between the number of sales calls made per day
# and the number of sales made per week. The company collected data on both variables from a sample of
# 30 sales representatives. Calculate the Pearson correlation coefficient between these two variables.

## **Q1. Pearson correlation coefficient is a measure of the linear relationship between two variables. Suppose you have collected data on the amount of time students spend studying for an exam and their final exam scores. Calculate the Pearson correlation coefficient between these two variables and interpret the result.**

In [11]:
# lest take an exmaple from the above question

# creating a dataset which cantains two variables one is a time students spents on studying for exam . and the another one is the result the get in the final exam
import pandas as pd
import numpy as np

df = pd.DataFrame({
    "time spent": [2,4,1,6,8],
    'results' : [50,40,30,60,80]
})

pearson_corr = df[['time spent','results']].corr(method='pearson').iloc[0,1]

print(f"pearson correlation coeficiant is  : {pearson_corr}")

pearson correlation coeficiant is  : 0.8986641371378515


### Certainly! The pearson correlation coefficient  is approximately 0.898. Here's a simple interpretation:

**Interpretation:**
The value of pearson correlation is close to 1, indicating a very strong positive relationship between the amount of students spent time on study and the final results the get. In simpler terms, as the amount of time the spent on study tends to higher , results tends to increase as well. This suggests a clear and consistent pattern: when students spents more time on studing , they are more likely to perform well in exam and get higher marks.

***

## **Q2. Spearman's rank correlation is a measure of the monotonic relationship between two variables. Suppose you have collected data on the amount of sleep individuals get each night and their overall job satisfaction level on a scale of 1 to 10. Calculate the Spearman's rank correlation between these two variables and interpret the result.**

In [12]:
import pandas as pd

# Create a DataFrame with the given data
data = {
    'Individual': [1, 2, 3, 4, 5],
    'Sleep': [7, 6, 8, 5, 7],
    'Job_Satisfaction': [8, 6, 9, 5, 7]
}

df = pd.DataFrame(data)

# Calculate spearman rank correlation 
# df.corr(method='spearman') 

# Calculate spearman rank correlation  coeficiant

rho = df[['Sleep','Job_Satisfaction']].corr(method='spearman').iloc[0,1]
print(f"spearman's rank correlation coeficiant (rho) is : {rho}")

spearman's rank correlation coeficiant (rho) is : 0.9746794344808964


Certainly! The Spearman's rank correlation coefficient (\( \rho \)) is approximately 0.975. Here's a simple interpretation:

**Interpretation:**
The value of rho  is close to 1, indicating a very strong positive monotonic relationship between the amount of sleep individuals get each night and their overall job satisfaction level. In simpler terms, as the amount of sleep increases, job satisfaction tends to increase as well. This suggests a clear and consistent pattern: when people sleep more, they are more likely to report higher levels of job satisfaction.

***

## **Q3. Suppose you are conducting a study to examine the relationship between the number of hours of exercise per week and body mass index (BMI) in a sample of adults. You collected data on both variables for 50 participants. Calculate the Pearson correlation coefficient and the Spearman's rank correlation between these two variables and compare the results.**

In [24]:
import pandas as pd
import numpy as np

# Set a random seed for reproducibility
np.random.seed(42)

# Generate random data for the number of hours of exercise per week and BMI
n_participants = 50

exercise_hours = np.random.uniform(2, 10, n_participants)  # Random hours of exercise between 2 and 10
bmi = 25 + 2 * exercise_hours + np.random.normal(0, 1, n_participants)  # BMI based on a linear relationship with exercise

# Create a DataFrame
data = {
    'Participant_ID': range(1, n_participants + 1),
    'Exercise_Hours': exercise_hours,
    'BMI': bmi
}

df = pd.DataFrame(data)

# Display the first few rows of the DataFrame
df.head()

# now we have calculate the spearman and pearson correlation coefficiant
pearson_corr = df[['Exercise_Hours','BMI']].corr(method='pearson').iloc[0,1]
spearman_corr = df[['Exercise_Hours','BMI']].corr(method='spearman').iloc[0,1]
print(f"the pearson's correlation coeficiant is : {pearson_corr} and spearman's rank correlation coeficiant is: {spearman_corr}")

the pearson's correlation coeficiant is : 0.9803763544334108 and spearman's rank correlation coeficiant is: 0.9776230492196879


**Interpretation:**
The code calculates both Pearson's correlation coefficient (\(r\)) and Spearman's rank correlation coefficient (\(\rho\)) using the synthetic dataset on the relationship between the number of hours of exercise per week and body mass index (BMI) for 50 participants.

1. **Pearson's Correlation Coefficient :**
   - pearson_corr = 0980
   - Interpretation: The positive value of pearson_corr indicates a strong positive linear relationship between the number of hours of exercise per week and BMI. In simpler terms, as the number of hours of exercise increases, BMI tends to increase as well.

2. **Spearman's Rank Correlation Coefficient :**
   - spearman_corr =0.977
   - Interpretation: The positive value of spearman_corr suggests a strong positive monotonic relationship between the two variables. This means that there is a consistent trend that as the number of hours of exercise per week increases, BMI tends to increase.

In summary, both correlation coefficients indicate a positive relationship between exercise hours and BMI in the provided dataset. The choice between Pearson and Spearman depends on the nature of the relationship and the distribution of the data. The high values of both coefficients suggest a clear trend, whether linear or monotonic, between the two variables in this synthetic dataset.

***

## **Q4. A researcher is interested in examining the relationship between the number of hours individuals spend watching television per day and their level of physical activity. The researcher collected data on both variables from a sample of 50 participants. Calculate the Pearson correlation coefficient between these two variables.**

In [9]:
# examining the relationship between  number of hours indivual spends on watching television per day and their level of physical activity 
# sample of 50
import pandas as pd



hours_watching_tv = [3, 4, 2, 5, 6, 1, 4, 3, 2, 5, 6, 1, 4, 3, 2, 5, 6, 1, 4, 3,
                     2, 5, 6, 1, 4, 3, 2, 5, 6, 1, 4, 3, 2, 5, 6, 1, 4, 3, 2, 5,
                     6, 1, 4, 3, 2, 5, 6, 1, 4, 3]

physical_activity = [150, 120, 200, 100, 80, 220, 130, 150, 180, 90, 75, 230,
                     120, 150, 190, 100, 80, 220, 130, 150, 180, 90, 75, 230, 120,
                     150, 190, 100, 80, 220, 130, 150, 180, 90, 75, 230, 120, 150,
                     190, 100, 80, 220, 130, 150, 180, 90, 75, 230, 120, 150]

len(hours_watching_tv)
len(physical_activity)
df = pd.DataFrame({'hours_watching_tv':hours_watching_tv,
                   'physical_activity':physical_activity})
df.corr(method='pearson')

hours_watching_tv  physical_activity
hours_watching_tv           1.000000          -0.988289
physical_activity          -0.988289           1.000000

### **the person correlation coeficient is 0.10173 and the correlation between the indiviadual spent hours to watch tv and thier physical activity is negative that means if the individuals spent more hours watching tv their physical activity is decreasing.**